internal/invocation/invocation_service.go

```diff
@@ -42,30 +45,40 @@ type Service struct {
 	urgentRequestCh chan Invocation
 	responseCh      chan *proto.ClientMessage
 	// removeCh carries correlationIDs to be removed
-	removeCh    chan int64
-	doneCh      chan struct{}
-	invocations map[int64]Invocation
-	handler     Handler
-	logger      ilogger.Logger
-	state       int32
+	removeCh        chan int64
+	doneCh          chan struct{}
+	groupLostCh     chan *GroupLostEvent
+	invocations     map[int64]Invocation
+	handler         Handler
+	eventDispatcher *event.DispatchService
+	logger          ilogger.Logger
+	state           int32
 }
 
 func NewService(
 	handler Handler,
+	eventDispacher *event.DispatchService,
 	logger ilogger.Logger) *Service {
-	service := &Service{
+	s := &Service{
 		requestCh:       make(chan Invocation),
 		urgentRequestCh: make(chan Invocation),
 		responseCh:      make(chan *proto.ClientMessage),
 		removeCh:        make(chan int64),
 		doneCh:          make(chan struct{}),
+		groupLostCh:     make(chan *GroupLostEvent, 1),
 		invocations:     map[int64]Invocation{},
 		handler:         handler,
+		eventDispatcher: eventDispacher,
 		logger:          logger,
 		state:           ready,
 	}
-	go service.processIncoming()
-	return service
+	s.eventDispatcher.Subscribe(EventGroupLost, serviceSubID, func(event event.Event) {
+		go func() {
+			s.groupLostCh <- event.(*GroupLostEvent)
```

Comment:
If we are doing this in a goroutine ? Is there a need for `groupLostCh` to have a buffer of 1 ? Can it be removed as well ?

Reply:
Yes, I've removed that at: https://github.com/hazelcast/hazelcast-go-client/pull/684/commits/d288c52993c2e3029d9eb3c9d73f455133b205d8